tests/test_misc.py

```diff
@@ -144,3 +145,37 @@ def test_unordered_dict(init_func):
                'confluent.monitoring.interceptor.topic': 'confluent-kafka-testing',
                'confluent.monitoring.interceptor.icdebug': False
                })
+
+
+# global variable for on_delivery call back function
+seen_delivery_cb = False
+
+
+def test_topic_config_update():
+    confs = [{"message.timeout.ms": 600000, "default.topic.config": {"message.timeout.ms": 1000}},
+             {"message.timeout.ms": 1000}, {"default.topic.config": {"message.timeout.ms": 1000}}]
+
+    def on_delivery(err, msg):
+        # Since there is no broker, produced messages should time out.
+        global seen_delivery_cb
+        seen_delivery_cb = True
+        assert err.code() == confluent_kafka.KafkaError._MSG_TIMED_OUT
+
+    for conf in confs[:]:
+        p = confluent_kafka.Producer(conf)
+
+        start = time.time()
+
+        timeout = start + 1
+
+        p.produce('mytopic', value='somedata', key='a key', on_delivery=on_delivery)
+        while time.time() < timeout:
+            if seen_delivery_cb:
+                break
+            p.poll(1)
+
+        duration = time.time() - start
+        if 1.02 >= duration <= .98:
```

Comment:
this is too tight. Since the "other" timeout is a lot higher (1 minute or 5 minutes), I think any value within 10s is okay here, which also makes it solid on CI.